src/smt/command.cpp

```diff
@@ -699,6 +699,7 @@ const api::Grammar* SynthFunCommand::getGrammar() const { return d_grammar; }
 
 void SynthFunCommand::invoke(api::Solver* solver, SymbolManager* sm)
 {
+  sm->addFunctionToSynthesize(d_fun);
```

Comment:
Why is it called here rather than when the command is declared?

Reply:
Creating a command should not do this.  It is only if/when the command is invoked that it should update the state of symbol manager.

Reply:
I see, cool.